oauth/client.go

```diff
@@ -0,0 +1,83 @@
+// Package oauth provides the OAuth APIs
+package oauth
+
+import (
+  "net/http"
+  "bytes"
+  "net/url"
+  stripe "github.com/stripe/stripe-go"
+)
+
+// Client is used to invoke /oauth and related APIs.
+type Client struct {
+  B   stripe.Backend
+  Key string
+}
+
+func AuthorizeURL(params *stripe.AuthorizeURLParams) string {
+  return getC().AuthorizeURL(params)
+}
+
+func (c Client) AuthorizeURL(params *stripe.AuthorizeURLParams) string {
+  var buf bytes.Buffer
+  buf.WriteString(stripe.CONNECTURL)
+
+  if stripe.BoolValue(params.Express) {
+    buf.WriteString("/express/oauth/authorize")
+  } else {
+    buf.WriteString("/oauth/authorize")
```

Comment:
I don't think we use `WriteString` in the rest of the code. We use `stripe.FormatURLPath` instead: https://github.com/stripe/stripe-go/blob/master/card/client.go#31 which prepends `api.stripe.com` but we should find a way to re-use this logic. This avoids escaping issues and such, especially for the GET params, which is a new concept.

Reply:
Switched to use FormatURLPath, but hacked it a little to get the connect domain and fit with `/express` because `/express` url encodes to `%2fexpress`

Reply:
Switched to using out form module to build the URL query string from a params struct.